src/classes/fillingDynamicallyCountrStateDistrictMap.cls

```diff
@@ -0,0 +1,151 @@
+        List<String> combinationOfCountryStateAndDistrict = new list<String>();
+        
+        //Check for Validate Conditions
+        If(!districtList.isEmpty()) {
+            for(District__c district : districtList) {
+                If(!mapofCountryStatesandDistricts.containsKey(district.State_Name__r.Country_Name__r.Name)) {
+                    mapofCountryStatesandDistricts.put(district.State_Name__r.Country_Name__r.Name, new Map<String, List<String>>());
+                }
+                If(!mapofCountryStatesandDistricts.get(district.State_Name__r.Country_Name__r.Name).containsKey(district.State_Name__r.Name)) {
+                    mapofCountryStatesandDistricts.get(district.State_Name__r.Country_Name__r.Name).put(district.State_Name__r.Name, new List<String>());
+                }
+                If(!mapofCountryStatesandDistricts.get(district.State_Name__r.Country_Name__r.Name).get(district.State_Name__r.Name).contains(district.Name)) {
+                    mapofCountryStatesandDistricts.get(district.State_Name__r.Country_Name__r.Name).get(district.State_Name__r.Name).add(district.Name);
+                  
+                    If(district.State_Name__r.Country_Name__r.Name.startsWith(firstLetter) && district.State_Name__r.Name.startsWith(firstLetter) &&
+                        district.Name.startsWith(firstLetter))                        
+                        combinationOfCountryStateAndDistrict.add(district.State_Name__r.Country_Name__r.Name + ' -> ' + district.State_Name__r.Name + 
+                        ' -> ' + district.Name);
+                }
+            }
+        }
+
+        System.debug('Map of District State and Country is:'+mapofCountryStatesandDistricts);
+        System.debug('Combination Key Set Of Country State and District is: '+combinationOfCountryStateAndDistrict);
+        aggregateCountDistrictandState();
+        countryJSONdata('Bharat');
+
+    }    
+
+    /**   Description :   Method To update records
+    *     Created By  :   Sudheer Kumar Shakya
+    *     Arguments   :   None
+    *     Return      :   Void
+    *     Revision log:   V_1.0 - Created  - Sudheer Kumar Shakya - Date - 15/07/2022
+    *                     V_1.1 - Modified - Sudheer Kumar Shakya - Date - 15/07/2022
+    **/
+    public static void aggregateCountDistrictandState(){
+
+        //Findng total Number of states using aggregate
+        Map<Id,State__c> mapofDistrictsinState=new Map<Id,State__c>();
+        AggregateResult[] districtAggregateResultForState = [Select Count(Id)Total_Districts, State_Name__c From District__c
+                                                             GROUP BY State_Name__c];
+        for(AggregateResult ar : districtAggregateResultForState) {
+            State__c state = new State__c();
+            state.Id = (Id)ar.get('State_Name__c');
+            state.Total_Number_of_Districts__c = (Integer)ar.get('Total_Districts');
+            mapofDistrictsinState.put(state.Id,state);
```

Comment:
remove line 67 to 70 and use below code and do same changes in below method.
mapofDistrictsinState.put((Id)ar.get('State_Name__c'), new State__c(Id = (Id)ar.get('State_Name__c'), Total_Number_of_Districts__c = (Integer)ar.get('Total_Districts')));